components/brave_vpn/common/wireguard/wireguard_utils.cc

```diff
@@ -82,6 +87,87 @@ WireguardKeyPair GenerateNewX25519Keypair() {
       EncodeBase64(std::vector<uint8_t>(privkey, privkey + 32)));
 }
 
+bool ValidateKey(const wchar_t* key,
+                 std::string* output,
+                 const std::string field_name) {
+  if (!base::WideToUTF8(key, wcslen(key), output)) {
+    VLOG(1) << "failed decoding " << field_name;
+    return false;
+  }
+
+  base::TrimWhitespaceASCII(*output, base::TRIM_ALL, output);
+
+  if ((*output).length() == 0) {
+    VLOG(1) << field_name << " does not have a value";
+    return false;
+  }
+
+  std::string decoded_config;
+  if (!base::Base64Decode(*output, &decoded_config) || decoded_config.empty()) {
+    VLOG(1) << field_name << " is not base64 encoded";
+    return false;
+  }
+
+  if (decoded_config.length() != 32) {
+    VLOG(1) << field_name << " is not the correct length";
+    return false;
+  }
+
+  return true;
+}
+
+bool ValidateAddress(const wchar_t* address, std::string* output) {
+  if (!base::WideToUTF8(address, wcslen(address), output)) {
+    VLOG(1) << "failed decoding address";
+    return false;
+  }
+
+  base::TrimWhitespaceASCII(*output, base::TRIM_ALL, output);
+
+  auto parsed = net::IPAddress::FromIPLiteral(*output);
+  if (!parsed.has_value()) {
+    VLOG(1) << "failed parsing address";
+    return false;
+  }
+
+  auto parsed_ip = parsed.value();
+  if (!parsed_ip.IsValid()) {
+    VLOG(1) << "address is not valid";
+    return false;
+  }
+
+  if (!parsed_ip.IsIPv4()) {
+    VLOG(1) << "address must be IPv4";
+    return false;
+  }
+
+  return true;
+}
+
+bool ValidateEndpoint(const wchar_t* endpoint, std::string* output) {
+  if (!base::WideToUTF8(endpoint, wcslen(endpoint), output)) {
+    VLOG(1) << "failed decoding endpoint";
+    return false;
+  }
+
```

Comment:
For all the fields Endpoint//Address//Key, I'd verify if they are defined in their alphabet with a regex, before passing to any parsing method.

- For Addresses `^[A-z0-9._-:[\]]+$` (I don't think we support empty IPs here)
- For Endpoints `^[A-z0-9._-:]+$` (I don't think we support empty endpoints here)
- For Keys `^[-A-Za-z0-9+/=]+$` (I don't think we support empty keys here)

Reply:
Added this validation in https://github.com/brave/brave-core/pull/24261/commits/624f5db9eabda65466e921d8836df6c8091b7921